src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingUtils.java

```diff
@@ -96,4 +106,43 @@ public static boolean logErrors(Map<String, String> sfConnectorConfig) {
   public static String getDlqTopicName(Map<String, String> sfConnectorConfig) {
     return sfConnectorConfig.getOrDefault(ERRORS_DEAD_LETTER_QUEUE_TOPIC_NAME_CONFIG, "");
   }
+
+  /**
+   * Get Approximate size of Sink Record which we get from Kafka. This is useful to find out how
+   * much data(records) we have buffered per channel/partition.
+   *
+   * <p>This is an approximate size since there is no API available to find out size of record.
+   *
+   * @param kafkaSinkRecord sink record received as is from Kafka (With connector specific converter
+   *     being invoked)
+   * @return long size of record in bytes.
+   */
+  public static long getApproxSizeOfRecordInBytes(SinkRecord kafkaSinkRecord) {
```

Comment:
This is technically not complete. I should rather do `GraphLayout.parseInstance(kafkaSinkRecord).totalSize()` to get size of entire object..

Reply:
I will change this once we know which library to use.